critical_section.c

```diff
@@ -0,0 +1,24 @@
+#include <critical_section.h>
+#include <interrupts.h>
+#include <common.h>
+#include <libkern.h>
+
+volatile int cs_level = 0;
```

Comment:
This `volatile` is redundant (and, actually, harmful). Disabling interrupts should already sufficiently protect the accesses to the variable on single-core platforms and on multi-core systems it is not enough anyway. So the only thing that `volatile` here does is that it, needlessly, prevents compiler optimisations.